eslint.config.js

```diff
@@ -0,0 +1,160 @@
+import comments from "@eslint-community/eslint-plugin-eslint-comments/configs";
+import eslint from "@eslint/js";
+import stylisticTs from "@stylistic/eslint-plugin-ts";
+import jsdoc from "eslint-plugin-jsdoc";
+import jsonc from "eslint-plugin-jsonc";
+import markdown from "eslint-plugin-markdown";
+import n from "eslint-plugin-n";
+import noOnlyTests from "eslint-plugin-no-only-tests";
+import perfectionist from "eslint-plugin-perfectionist";
+import regexp from "eslint-plugin-regexp";
+import vitest from "eslint-plugin-vitest";
+import yml from "eslint-plugin-yml";
+import globals from "globals";
+import tseslint from "typescript-eslint";
+
+export default tseslint.config(
+	// Global ignores
+	{
+		ignores: [
+			"!.*",
+			"coverage",
+			"dist",
+			"lib",
+			"node_modules",
+			"pnpm-lock.yaml",
+		],
+	},
+	{ linterOptions: { reportUnusedDisableDirectives: "error" } },
+
+	eslint.configs.recommended,
+	comments.recommended,
+	n.configs["flat/recommended"],
+	perfectionist.configs["recommended-natural"],
+	regexp.configs["flat/recommended"],
+	vitest.configs.recommended,
+	markdown.configs.recommended,
+
+	{
+		files: ["**/*.{js,cjs,mjs,ts,cts,mts}"],
+		languageOptions: {
+			globals: {
+				...globals.es2022,
+				...globals.node,
+			},
+			parser: tseslint.parser,
+		},
+		plugins: {
+			"@stylistic/ts": stylisticTs,
+			"@typescript-eslint": tseslint.plugin,
+			jsdoc: jsdoc,
+			"no-only-tests": noOnlyTests,
+		},
+		rules: {
+			// These off/less-strict-by-default rules work well for this repo and we like them on.
+			"@typescript-eslint/no-unused-vars": ["error", { caughtErrors: "all" }],
+			"no-only-tests/no-only-tests": "error",
```

Comment:
I've stopped using the `no-only-tests` plugin because Vitest blocks `.only`s in CI by default: https://vitest.dev/config/#allowonly

```suggestion
```